File: StructuralDesignPattern/AdapterPattern/java/ConcreteTarget/Mp3Player.java
package StructuralDesignPattern.AdapterPattern.java.ConcreteTarget;

import StructuralDesignPattern.AdapterPattern.java.AbstractTarget.MediaPlayer;
import StructuralDesignPattern.AdapterPattern.java.Adapter.MediaAdapter;

public class Mp3Player implements MediaPlayer{

    MediaAdapter mediaAdapter;

    @Override
    public void play(String audioType, String fileName) {
        // Built-in support for MP3
        if (audioType.equalsIgnoreCase("mp3")) {
            System.out.println("Playing MP3 file. Name: " + fileName);
        }
        // Use adapter for VLC and MP4 files
        else if (audioType.equalsIgnoreCase("vlc") || audioType.equalsIgnoreCase("mp4")) {
            mediaAdapter = new MediaAdapter(audioType);
            mediaAdapter.play(audioType, fileName);
        } else {
            System.out.println("Invalid media. " + audioType + " format not supported");
        }
    }
    
}

File: StructuralDesignPattern/AdapterPattern/python/ConcreteTarget/Mp3Player.py
from AbstractTarget.MediaPlayer import MediaPlayer
from Adapter.MediaAdapter import MediaAdapter


class Mp3Player(MediaPlayer):

    def __init__(self):
        self._mediaAdapter = None

    def play(self,mediaType,fileName):
        if mediaType == "mp3":
            print("Playing mp3 file {}".format(fileName))
        elif mediaType == "mp4":
            self._mediaAdapter = MediaAdapter("mp4")
            self._mediaAdapter.play(mediaType, fileName)
        elif mediaType  == "vlc":
            self._mediaAdapter = MediaAdapter("vlc")
            self._mediaAdapter.play(mediaType, fileName)
        else:
            print("Not supported")